sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -192,39 +196,214 @@ private void sendBody(HttpRequest httpRequest, Object progressReporter, HttpURLC
      *
      * @param httpRequest The HTTP Request being sent
      * @param connection The HttpURLConnection being sent to
-     *
      * @return A HttpResponse object
      */
     private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection connection) {
         try {
             int responseCode = connection.getResponseCode();
+            Headers responseHeaders = getResponseHeaders(connection);
+            ServerSentEventListener listener = httpRequest.getServerSentEventListener();
+            RetrySSEResult retrySSEResult;
+            if (connection.getErrorStream() == null) {
+                if (isTextEventStream(responseHeaders) && listener != null) {
```

Comment:
What happens if this is a text event stream but the listener is `null`? Will we read the whole response into memory?

Reply:
We could just log and let the user know that there was no listener attached and skip reading the data. As in-memory data without the listener won't reach the user anyway.